src/client/service-worker/events/fetch.js

```diff
@@ -23,8 +23,50 @@ import {
 } from '@americanexpress/one-service-worker';
 
 import { OFFLINE_CACHE_NAME } from '../constants';
+import { createResourceMetaData, fetchCacheResource } from './utility';
+
+function createAppCachingMiddleware(oneAppVersion) {
+  return function appResourceCachingMiddleware(event, context) {
+    if (event.request.url.includes(oneAppVersion)) {
+      const [baseAppUrl] = event.request.url.split(oneAppVersion);
+      const meta = createResourceMetaData(
+        event,
+        ['app', `${baseAppUrl}${oneAppVersion}/`]
+      );
+      context.set('cacheName', meta.cacheName);
+      context.set('request', event.request.clone());
+      event.respondWith(fetchCacheResource(event, meta));
+    }
+  };
+}
+
+function createHolocronCachingMiddleware(holocronModuleMap) {
+  const { clientCacheRevision, modules } = holocronModuleMap;
+  const moduleNames = Object.keys(modules);
+  const moduleEntries = moduleNames.map((name) => [name, modules[name].baseUrl]);
+
+  return function holocronCachingMiddleware(event, context) {
+    // TODO: optimize - revise algorithm for matching
+    const matchingModule = moduleEntries
+      .find(([, baseUrl]) => event.request.url.startsWith(baseUrl));
+
+    if (matchingModule) {
+      const meta = createResourceMetaData(
+        event,
+        matchingModule,
+        clientCacheRevision
+      );
+      context.set('cacheName', meta.cacheName);
+      context.set('request', event.request.clone());
+      event.respondWith(fetchCacheResource(event, meta));
+    }
+  };
+}
 
 export default function createFetchMiddleware() {
+  const oneAppVersion = process.env.ONE_APP_BUILD_VERSION;
+  const holocronModuleMap = JSON.parse(process.env.HOLOCRON_MODULE_MAP);
```

Comment:
`process.env.HOLOCRON_MODULE_MAP` Where does this come from? Having trouble finding this. Where is the documentation on these ENV vars? https://one-amex-docs.americanexpress.com/en-us/one-app/api/server/environment-variables

Reply:
Both `ONE_APP_BUILD_VERSION` and `HOLOCRON_MODULE_MAP` are injected values; `ONE_APP_BUILD_VERSION` is [replaced during build time ](https://github.com/americanexpress/one-app/blob/feature/pwa-cache/scripts/build-service-workers.js#L31) and the `HOLOCRON_MODULE_MAP` is injected in the [service worker middleware](https://github.com/americanexpress/one-app/blob/feature/pwa-cache/src/server/middleware/pwa/service-worker.js#L21-L24).

Reply:
Could we place this information in a well thought out README? This process is rather unusual given the nature of merging a PWA-only build system into Server middleware.

Reply:
Updated with additional docs breaking down details (f21a72c) (a4525b7)